src/melee.cpp

```diff
@@ -1715,8 +1712,10 @@ std::string player::melee_special_effects(Creature &t, damage_instance &d, const
 
     handle_melee_wear();
 
-    bool is_hallucination = false; //Check if the target is an hallucination.
-    if(monster *m = dynamic_cast<monster *>(&t)) { //Cast fails if the t is an NPC or the player.
+    // Check if the target is an hallucination.
+    bool is_hallucination = false;
+    monster *m = dynamic_cast<monster *>(&t);
+    if( m != nullptr ) {
```

Comment:
I'll close this one then. It's a tiny fix - just one line really needs to be changed.